statesync/syncer.go

```diff
@@ -313,6 +315,37 @@ func (s *syncer) Sync(snapshot *snapshot, chunks *chunkQueue) (sm.State, *types.
 	return state, commit, nil
 }
 
+// LocalSync restores stateStore and blockStore state which is not included in the snapshot
+// This function assumes that the implicit trusted local snapshot chunks have already been applied
+// State restoration depends on heights after the restored snapshot so the RPC dependency remains
+func (s *syncer) LocalSync() (sm.State, *types.Commit, error) {
+	pctx, pcancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer pcancel()
+
+	// Optimistically build new state, so we don't discover any light client failures at the end.
+	state, err := s.stateProvider.State(pctx, s.restoreHeight)
+	if err != nil {
```

Comment:
does this still use RPCs? I think we can go one step further and verify against local state cause we can safely assume we have the hashes and a block within the unbnonding period.

Reply:
@cmwaters do you happen to know?

Reply:
Yeah you're still going to need to use RPC

Reply:
hmm mentioned below but there should be a way to do this with out (https://github.com/tendermint/tendermint/issues/4642)

Reply:
Looked at https://github.com/tendermint/tendermint/issues/4642 and seems to only recommend adding support, but doesn't provide a method or direction.